### ISSUE TITLE
Parse on a 'refinement' fails

### ISSUE THREAD

**Issue by rgchris** [metaeducation/ren-c#1030]
Parsing on refinements fails in the current build:

```
Rebol 3 (Ren-C branch) [version: 2.102.0.2.40 build: 16-Nov-2019/8:07:15+0:00]

>> parse [/foo][/foo]
; null
```

This seems problematic where the 'refinement' style of path is still a first-class dialect element within the core language (function specifications).

**Comment by hostilefork** [comment]
> I understand you can handle this with a lit-bit, but is still worth consideration.

I'm sure similar arguments would seem to apply to **parse [2] [2]**...but don't.

There's been a lot to absorb about the concept of embracing **/foo** as a PATH!...and unfortunately not enough time and testing has been spent on it.  I do think it's a solid decision that just hasn't gotten its due.  The dialect abilities to say **/foo/** or **foo/** or **/foo/bar** may find their way into meaningful applications in function specs, even.

On the raised point: I've worried that with **'/foo** and **@/foo** we may not be doing ourselves any favors by having /foo not be evaluator-active in people's minds.  Having it be a PATH! and actually meaning something w.r.t. binding behavior or lookup might be a stronger asset.  I don't know what it would mean, but in C++ when you use scope resolution like **namespace::class::method** you can leave off the namespace and say **::thing** and that means to "unscope" the item.  I'm not sure what "unscoping" would be in Rebol, but it could (for instance) be something functions ignore rebinding of but modules don't...

    Rebol [
       Name: "Path active demo"
       Type: 'module
    ]

    foo: 10

    bar: func [foo] [
        foo: 20
        print ["foo is" foo "/foo is" /foo]
    ]

Maybe that would get you **foo is 20 /foo is 10**?  I don't know.  But it's the kind of possibility opened up when **/foo** isn't a reduced form, but able to do other things like **:/foo/bar**.

So long story short: I think the transition to PATH! is a nice strong direction, and underscores that the PATH!-ness is probably the more important tinkertoy part here than the "looks like what used to be an inert word type" aspect.

**Comment by hostilefork** [comment]
With the passage of time, I think that **/foo** and **/foo/bar** being evaluator inactive is probably for the best.  (I don't know what **/foo:** or **:/foo** will do, however.)

**But this further points to how it might go wasted in PARSE if used as a simple synonym for '/foo.**  Consider how generic TUPLE! is used for predicates, and having that in PARSE sounds very nice:

    >> parse [2 4] [some .not.odd?]
    == [2 4] 

[(Note progress is available still as the /PROGRESS output, second multi-return.)](https://forum.rebol.info/t/the-parse-of-progress/1349)

The reason TUPLE! works for predicate can do this is because it's a datatype that carries bindings, and that you know isn't intended as a variable reference since it would be inert.

So what might the parallel be for /foo/bar in dialecting terms?  (Refinements were originally thought of as being used for predicates, but the dots seemed nicer.)

     >> parse [...] [some /foo/bar]
     ; == ??

It even provides another BLOCK! and GROUP! form if you want, **/(...)** or **/[...]**.  This might be useful for the "use literally" idea.

    >> data: [some integer!]

    >> parse [[some integer!] [some integer!]] [2 /data]
    == [[some integer!] [some integer!]]

That would free up the @(...) forms for "literally, this" to things like KEEP.

    >> parse [1 2] [collect x [some [keep @[<a> <b>], keep @(reverse "edoc"), keep integer!]]]
    >> x
    == [<a> <b> "code" 1 <a> <b> "code" 2]

Random ideas, but the point is that there's a lot more power available with this bound type than just a synonym for its quoted literal form.

> There's been a lot to absorb about the concept of embracing /foo as a PATH!...and unfortunately not enough time and testing has been spent on it. I do think it's a solid decision that just hasn't gotten its due.

The more I work with it the more certain I am that this is the right...path...to take.
